### PR TITLE
Adding tanh activation function for torch, addresses the issue #2686

### LINKED ISSUE #2686: tanh

### PR DESCRIPTION
**Pull request by bsovdat**
Fixed some issues...

closes #2686

### CODE REVIEW

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Great work so far! Please remove any changes related to the `idea` folder and only commit or keep changes related to the function and tests. As usual, please request my review again once you're done! Thank you!

**Review by bsovdat — COMMENTED** [review]
Now i am only commiting .py files?!

File: ivy/functional/frontends/torch/non_linear_activation_functions.py [hunk 1]
```diff
@@ -34,3 +33,9 @@ def gelu(input, approximate="none"):
 
 
 gelu.unsupported_dtypes = ("float16",)
+
+def tanh(input, *, out=None):
+    return ivy.tanh(input, out=out)
+
+
+tanh.unsupported_dtypes = ("float16",)
```
Comment (juliagsy) [inline ivy/functional/frontends/torch/non_linear_activation_functions.py:41]: this is only unsupported in torch, please use `{"torch": ("float16",)}` instead

File: ivy_tests/test_ivy/test_frontends/test_torch/test_non_linear_activation_functions.py [hunk 2]
```diff
@@ -175,3 +175,43 @@ def test_torch_leaky_relu(
...
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=tuple(
+            set(ivy_np.valid_float_dtypes).intersection(
+                set(ivy_torch.valid_float_dtypes)
+            )
+        )
+    ),
+    as_variable=st.booleans(),
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_non_linear_activation_functions.py:189]: please remove `as_variable`, `with_out` and `native_array` here, they are generated internally through the `@handle_cmd_line_arg` decorator

but remain to **keep** them in the argument lists in `test_torch_tanh`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_non_linear_activation_functions.py [hunk 3]
```diff
@@ -175,3 +175,43 @@ def test_torch_leaky_relu(
...
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    input_dtype, x = dtype_and_x
+
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_non_linear_activation_functions.py:207]: please use `[input_dtype]` here

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Thanks for the previous changes! There are some changes required before the PR is good to go! As usual, please request my review once you're done! Thank you!

**Review by juliagsy — APPROVED** [review]
Thanks for contributing! I will merge the PR soon!

**Pull request merged**